### DISCUSSION TITLE
dynamically setting contract deployed address

### DISCUSSION THREAD

**Discussion by eden91940** [wevm/wagmi#719] [Q&A]
Hello, 

I'm looking for the best to set dynamically the  addressOrName of my deployed contract in a useContract  awaiting that : 

```
  const networkId = provider.network.chainId;
    let deployedNetwork = MyABI.networks[networkId];
    console.log(deployedNetwork.address)
```

Thx

**Comment by nasa8x** [comment]
```js
 const networkId = provider.network.chainId;
 let deployedNetwork = MyABI.networks[networkId];

 const contract = useContract({
    addressOrName: deployedNetwork.address
  })
```

**Reply by eden91940** [reply]
Ive got this after a truffle migrate withe a ganache chainId 1337, it doesnt match 

  "networks": {
    "1658388536973": {
      "events": {},
      "links": {},
      "address": "0x3568c7938E548806b55eA9eCD317f65aD347EE9D",
      "transactionHash": "0x2c5758bd371bbdb1fb54e6222957fbade2dcf8584201efdf8e2efe47950b02ed"
    }
  }

I want to wait my for my networkPromise = > 

`
export function useVotingContract() {

    const provider = useProvider();
 
    let networkSelected;
    provider.getNetwork().then((network) => {
        networkSelected = network
    })

    const config = {
        addressOrName: VotingABI.networks[networkSelected.chainId].address,
        contractInterface: VotingABI,
    };

    const contractProvider = useContract({
        ...config,
        signerOrProvider: provider,
    }, [config]);

    const contractSigner = useContract({
        ...config,
        signerOrProvider: provider.getSigner(),
    });


    return {contractConfig: config, contractProvider, contractSigner};

}`

same problem with signer...
